docs/guidance_for_upgraders.md

```diff
@@ -0,0 +1,74 @@
+# Guidance for Operator Developer
```

Comment:
Overall, I think there are some grammar issues in this doc, could you try copy them into a google doc, which will detect these issues for you to fix.

Reply:
Just did it and it seems fine. Will check again after the last commit